oss-internship-2020/libtiff/example/main_sandboxed.cc

```diff
@@ -0,0 +1,277 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cstring>
+#include <iostream>
+#include <vector>
+
+#include "../sandboxed.h"
+#include "sandboxed_api/sandbox2/util/fileops.h"
+#include "sandboxed_api/sandbox2/util/path.h"
+#include "tiffio.h"
+
+/* sapi functions:
+ *    TIFFTileSize
+ *    TIFFOpen
+ *    TIFFReadEncodedTile
+ *    TIFFSetField
+ *    TIFFClose
+ *    TIFFReadRGBATile
+ *    TIFFGetField
+ */
+
+static const std::vector<unsigned char> cluster_0 = {0, 0, 2, 0, 138, 139};
+static const std::vector<unsigned char> cluster_64 = {0, 0, 9, 6, 134, 119};
+static const std::vector<unsigned char> cluster_128 = {44, 40, 63, 59, 230, 95};
```

Comment:
Alright, there are multiple comments here:

1. Even though those `std::vector`s are marked `const`, they are nevertheless not-trivally destructible, which is prohibited by our style guide (see https://google.github.io/styleguide/cppguide.html#Static_and_Global_Variables).
  In C++20, we will be able to write `constexpr std::vector<uint8_t> kCluster0 = {...}`, but right now, we are stuck with C++17.<br>
  One way to deal with this is to simply use pointers that are never freed (the OS will eventually free them for us):
  ```c++
  static const auto* g_cluster0 = new std::vector<uint8_t>{0, 0, 2, ...};
  ```
  See https://g3doc.corp.google.com/devtools/library_club/g3doc/totw/140.md?cl=head<br>
  In this case the best way to declare these is probably using `std::array<>`, like this
  ```c++
  constexpr std::arrary<uint8_t, 6> kCluster0 = {0, 0, 2, 0, 138, 139};
  ```
2. Prefer standard integer types from `cstdint` to plain `unsigned char` and the like: `uint8_t` in this case.
3. These variables should be named `g_cluster0`, `g_cluster64` and `g_cluster128` if you keep the `std::vector` or (if you make them constant): `kCluster0`, etc.